backend/controllers/createLinkedDpsUser.test.js

```diff
@@ -0,0 +1,384 @@
+const { createLinkedDpsUserFactory } = require('./createLinkedDpsUser')
+
+describe('create linked user factory', () => {
+  const getCaseloads = jest.fn()
+  const createLinkedAdminUser = jest.fn()
+  const createLinkedLsaUser = jest.fn()
+  const createLinkedGeneralUser = jest.fn()
+  const searchUser = jest.fn()
+
+  const createLinkedDpsUser = createLinkedDpsUserFactory(
+    getCaseloads,
+    createLinkedAdminUser,
+    createLinkedLsaUser,
+    createLinkedGeneralUser,
+    searchUser,
+    '/create-linked-dps-user',
+    '/create-dps-user',
+    '/create-user',
+    '/manage-dps-users',
+  )
+
+  describe('index', () => {
+    it('should call create-user index scenario 1', async () => {
+      const req = {
+        query: { action: 'searchUser' },
+        flash: jest.fn().mockReturnValueOnce(''),
+      }
+      const render = jest.fn()
+      const redirect = jest.fn()
+      await createLinkedDpsUser.index(req, { render, redirect })
+
+      expect(req.flash).toBeCalledWith('user')
+      expect(redirect).toBeCalledWith('/create-linked-dps-user')
+    })
+
+    it('should call create-user index scenario 2', async () => {
+      const req = {
+        query: { action: '' },
+        flash: jest.fn().mockReturnValueOnce(''),
+      }
+      const render = jest.fn()
+      const redirect = jest.fn()
+      await createLinkedDpsUser.index(req, { render, redirect })
+
+      expect(req.flash).toBeCalledWith('user')
+      expect(redirect).toBeCalledWith('/create-user')
+    })
+
+    it('should call create-user index scenario 3', async () => {
```

Comment:
Please add actual scenario in the test (for above two tests)

Reply:
Done, thanks